src/main.rs

```diff
@@ -83,7 +96,81 @@ fn main() {
         Rank::InfoPlusProbability => roget::Rank::InfoPlusProbability,
         Rank::ExpectedInformation => roget::Rank::ExpectedInformation,
     };
-    play(move || solver.build(), args.games);
+    if args.interactive {
+        play_interactive(solver.build());
+    } else {
+        play(move || solver.build(), args.games);
+    }
+}
+
+fn play_interactive(mut guesser: impl Guesser) {
+    let mut history = Vec::with_capacity(6);
+    println!("C: Correct / Green, M: Misplaced / Yellow, W: Wrong / Gray");
+    // Wordle only allows six guesses.
+    for _ in 1..=6 {
+        let guess = guesser.guess(&history);
+        println!("Guess:  {}", guess.to_uppercase());
+        let correctness = {
+            loop {
+                match ask_for_correctness() {
+                    Ok(c) => break c,
+                    Err(e) => println!("{}", e),
+                }
+            }
+        };
+        if correctness == [roget::Correctness::Correct; 5] {
+            println!("The answer was {}", guess.to_uppercase());
+            return;
+        }
+        history.push(roget::Guess {
+            word: Cow::Owned(guess),
+            mask: correctness,
+        });
+    }
+    println!("Game Over, only six guesses are allowed");
+}
+
+fn ask_for_correctness() -> Result<[roget::Correctness; 5], Cow<'static, str>> {
+    print!("Colors: ");
```

Comment:
I think it's this prompt that should be more helpful. Something like:

> What colors did Wordle report for the guess? (C = correct, M = misplaced, W = wrong)

Reply:
The prompt is this short in order to be aligned with the guess above; making it this long would mean having the guess either unaligned with the colors, or having both of them at the end of a line, with wrapping if this is viewed on a thin display.

Reply:
That's a good point! What if we print out these instructions just once further up in the output (like before we even print the first word to guess)? That way we can have a longer description of the expected input without breaking the alignment.

Also, "colors" is a little weird here given that we take C/M/W as input, not G(reen)/Y(ellow)/G(ray).

Reply:
Ah, you already added instructions at the top, nice! I still think "Colors" is a little weird, but let's not block on that :+1: